http3/http_stream.go

```diff
@@ -268,3 +278,23 @@ func (s *stream) ReceiveDatagram(ctx context.Context) ([]byte, error) {
 	// TODO: reject if datagrams are not negotiated (yet)
 	return s.datagrams.Receive(ctx)
 }
+
+type warpReader struct {
+	io.Reader
+	first bool
+	cb    func()
+}
+
+func (r *warpReader) Read(b []byte) (int, error) {
+	if r.cb == nil || len(b) == 0 || r.first {
+		return r.Reader.Read(b)
+	}
+	n1, err := r.Reader.Read(b[:1])
+	if err != nil {
+		return 0, err
+	}
+	r.first = true
+	r.cb()
+	n2, err := r.Reader.Read(b[1:])
+	return n1 + n2, err
```

Comment:
This can be simplified. An `io.Reader` will return `n < len(b)` if less than `len(b)` bytes are available:
```suggestion
	n, err := r.Reader.Read(b)
	if n > 0 && !r.first {
		traceGotFirstResponseByte(s.trace)
		r.first = true
	}
	return n, err
}
```

Reply:
fixed